src/System.Management.Automation/engine/parser/VariableAnalysis.cs

```diff
@@ -954,6 +954,11 @@ public object VisitScriptBlock(ScriptBlockAst scriptBlockAst)
                 scriptBlockAst.EndBlock.Accept(this);
             }
 
+            if (scriptBlockAst.CleanupBlock != null)
+            {
+                scriptBlockAst.CleanupBlock.Accept(this);
+            }
+
             _currentBlock.FlowsTo(_exitBlock);
```

Comment:
Is this true in this case? If we visit `CleanupBlock` and set it as `_currentBlock`, does its output flow to `_exitBlock`? Or have I misinterpreted how the code works?

Reply:
I would trust your understanding of the code here over mine, any day, it's not code I'm super familiar with. 

On the assumption that `_exitBlock` actually refers to _end_ block, then no, absolutely not. In fact, if this refers to _output_ in the usual sense (pipeline/success output), then `cleanup{}` discards any output that code within it sends to the pipeline.

Reply:
@daxian-dbw understands this code best, worth us seeking an opinion from him.

> In fact, if this refers to output in the usual sense (pipeline/success output), then cleanup{} discards any output that code within it sends to the pipeline.

Yeah that's the thought going through my mind

Reply:
I could well be wrong, but I _think_ the `FlowsTo` is referring to logic flow rather than output flow? Not sure though.